frontend/components/ui/CopyButton.tsx

```diff
@@ -0,0 +1,30 @@
+import React, { useState } from 'react';
+
+const CopyButton = ({ textcode }: { textcode: string }) => {
+    const [copied, setCopied] = useState(false);
+    const cleanedCode = textcode.replace(/```js|```/g, '');
+
+    const copyToClipboard = () => {
+        navigator.clipboard.writeText(cleanedCode);
+        setCopied(true);
+
+        // Reset the "copied" state after 3 seconds
+        setTimeout(() => {
```

Comment:
Assign it to a constant so we can `clearTimeout` in the callback function @tanthemess